pkg/server/config.go

```diff
@@ -315,6 +316,7 @@ func NewConfig(opts *kcpserveroptions.CompletedOptions) (*Config, error) {
 		*c.preHandlerChainMux = append(*c.preHandlerChainMux, mux)
 		apiHandler = mux
 
+		apiHandler = revdial.WithSyncerTunnel(apiHandler)
```

Comment:
@s-urbaniak how can I add authentication to this handler?

Reply:
@aojea is authentication or is it authorization you want to add here? how and why should it be different from the default authz/authn we already configure in kcp?

Generally. your handler here already gets authz as described in https://github.com/kcp-dev/kcp/blob/main/docs/authorization.md.

When it comes to authentication it also gets the stock authentication scheme from within kcp.

Reply:
oh i see, this handler is before authorization is initialized, and as discussed OOB you cannot put it later in the chain. i suggest we discuss together with @sttts what we envision authz wise.

Reply:
thanks sur :+1: